exonum-java-binding/core/rust/integration_tests/tests/runtime_error_handling.rs

```diff
@@ -0,0 +1,194 @@
+        env.throw_new(ARITHMETIC_EXCEPTION_CLASS, EXPECTED_DESCRIPTION)?;
+        wait_for_thrown_exception(env)?;
+        Ok(())
+    });
+}
+
+#[test]
+fn jni_call_default_illegal_argument_exception() {
+    let expected = ErrorMatch::from_fail(&Error::IllegalArgument)
+        .with_description_containing(EXPECTED_DESCRIPTION);
+    assert_jni_call_default_errors(&EXECUTOR, expected, |env| {
+        env.throw_new(ILLEGAL_ARGUMENT_EXCEPTION_CLASS, EXPECTED_DESCRIPTION)?;
+        wait_for_thrown_exception(env)?;
+        Ok(())
+    });
+}
+
+#[test]
+fn jni_call_transaction_jni_error() {
+    let expected = ErrorMatch::from_fail(&Error::JniError)
+        .with_description_containing("Invalid constructor return type (must be void)");
+    assert_jni_call_transaction_errors(&EXECUTOR, expected, |env| {
+        // use invalid constructor signature to force JNI error
+        env.new_object(STRING_CLASS, "()I", &[])?;
+        Ok(())
+    });
+}
+
+#[test]
+fn jni_call_transaction_any_java_exception() {
+    let expected = ErrorMatch::from_fail(&Error::JavaException)
+        .with_description_containing(EXPECTED_DESCRIPTION);
+    assert_jni_call_transaction_errors(&EXECUTOR, expected, |env| {
+        env.throw_new(ARITHMETIC_EXCEPTION_CLASS, EXPECTED_DESCRIPTION)?;
+        wait_for_thrown_exception(env)?;
+        Ok(())
+    });
+}
+
+#[test]
+fn jni_call_transaction_illegal_argument_exception() {
+    let expected = ErrorMatch::from_fail(&Error::IllegalArgument)
+        .with_description_containing(EXPECTED_DESCRIPTION);
+    assert_jni_call_transaction_errors(&EXECUTOR, expected, |env| {
+        env.throw_new(ILLEGAL_ARGUMENT_EXCEPTION_CLASS, EXPECTED_DESCRIPTION)?;
+        wait_for_thrown_exception(env)?;
+        Ok(())
+    });
+}
+
+#[test]
+fn jni_call_transaction_transaction_execution_exception() {
+    let expected = ErrorMatch::from_fail(&FakeServiceError::Error)
+        .with_description_containing(EXPECTED_DESCRIPTION);
+    let err = jni_call_transaction(&EXECUTOR, |env| {
+        let exception = create_transaction_execution_exception(env)?;
+        env.throw(JThrowable::from(exception))?;
+        wait_for_thrown_exception(env)?;
+        Ok(())
+    })
+    .unwrap_err();
+    assert_eq!(err, expected);
+}
+
+#[test]
+fn jni_call_transaction_unexpected_transaction_execution_exception() {
+    let err = jni_call_transaction(&EXECUTOR, |env| {
+        let cause = create_any_java_exception(env)?;
+        let exception = env.new_object(
+            UNEXPECTED_TRANSACTION_EXECUTION_EXCEPTION_CLASS,
+            "(Ljava/lang/Throwable;)V",
+            &[cause.into()],
+        )?;
+        env.throw(JThrowable::from(exception))?;
+        wait_for_thrown_exception(env)?;
+        Ok(())
+    })
+    .unwrap_err();
+    assert_eq!(err.kind(), ErrorKind::Unexpected);
+    assert_eq!(err.description(), EXPECTED_DESCRIPTION);
+}
+
+fn create_transaction_execution_exception<'a>(env: &'a JNIEnv) -> JniResult<JObject<'a>> {
```

Comment:
There is no longer a transaction execution exception (abe31e590bf43df41a536609381f8eed126a3d12).

The method under test and tests might need to be renamed. However, we may still
use 'transaction' to distinguish it from 'default'.